libs/api/domains/driving-license/src/lib/graphql/main.resolver.ts

```diff
@@ -49,4 +48,71 @@ export class MainResolver {
   drivingLicensePenaltyPointStatus(@CurrentUser() user: User) {
     return this.drivingLicenseService.getPenaltyPointStatus(user.nationalId)
   }
+
+  @Query(() => HasTeachingRights)
+  drivingLicenseTeachingRights(@CurrentUser() user: User) {
+    return this.drivingLicenseService.getTeachingRights(user.nationalId)
+  }
+
+  @Query(() => StudentInformationResult)
+  drivingLicenseStudentInformation(@Args('nationalId') nationalId: string) {
+    const student = this.drivingLicenseService.getStudentInformation(nationalId)
+
+    return {
+      student,
+    }
+  }
+
+  @Query(() => ApplicationEligibility)
+  drivingLicenseApplicationEligibility(
+    @CurrentUser() user: User,
+    @Args('type') type: string,
+  ) {
+    return this.drivingLicenseService.getApplicationEligibility(
+      user.nationalId,
+      type,
+    )
+  }
+
+  @Query(() => [Juristiction])
+  drivingLicenseListOfJuristictions() {
+    return this.drivingLicenseService.getListOfJuristictions()
+  }
+
+  @Mutation(() => NewDrivingAssessmentResult)
+  async drivingLicenseNewDrivingAssessment(
+    @Args('input') input: NewDrivingAssessmentInput,
+    @CurrentUser() user: User,
+  ): Promise<NewDrivingAssessmentResult> {
```

Comment:
Should all of these queries and mutations be publicly accessible?

Unfortunately the current data provider setup in the application system "encourages" adding everything to the graphql API (since dataProviders and template modules both have a utility to send graphql requests). But sensitive/internal APIs should not be in the graphql API. Instead, we should refactor integrations/clients to `libs/clients` and expose some stuff in graphql, but call other stuff directly from the application system template-module.

Template modules can get other NestJS modules injected. We have on the roadmap to support the same in dataProviders.

Reply:
ah, i see. yes, we definitely don't want them publicly accessible.

ok, so you think i should add the driver's license nest module as a dependency of the api template, instead of calling the graphql API?

And then think through whether or not to expose all those endopints to graphql. Makes sense, if I understand correctly

Reply:
Yes exactly.

Reply:
ok, i updated the PR with this change, if I understood :)